File: backend/corelibs/seguridad.py
"""Operaciones sencillas de seguridad y hashing."""

import hashlib
import uuid


def hash_md5(texto: str) -> str:
    """Devuelve el hash MD5 de *texto*."""
    return hashlib.md5(texto.encode("utf-8")).hexdigest()


def hash_sha256(texto: str) -> str:
    """Devuelve el hash SHA-256 de *texto*."""
    return hashlib.sha256(texto.encode("utf-8")).hexdigest()


def generar_uuid() -> str:
    """Genera un identificador único."""
    return str(uuid.uuid4())

File: src/core/nativos/seguridad.js
import crypto from 'crypto';
import { v4 as uuidv4 } from 'uuid';

export function hash_md5(texto) {
    return crypto.createHash('md5').update(texto).digest('hex');
}

export function hash_sha256(texto) {
    return crypto.createHash('sha256').update(texto).digest('hex');
}

export function generar_uuid() {
    return uuidv4();
}
